TKGame/World/Components/C_World_LevelEditor.cs

```diff
@@ -23,28 +23,94 @@ void ILevelEditorComponent.Update()
                 LevelEditor.SaveStageDataToJSON(TKGame.levelComponent.GetCurrentStage(), saveName);
             }
+
+            #region Structure Selection
+            // Ctrl + 1 = Wall; Ctrl + 2 = Spikes; Ctrl + 3 = None
+            // TODO: add visual indicator for selected structure
+            if (Input.KeyboardState.IsKeyDown(Keys.LeftControl) && Input.WasKeyPressed(Keys.D1))
+            {
+                LevelEditor.selectedStructure = StructureType.Wall;
+            }
+            else if (Input.KeyboardState.IsKeyDown(Keys.LeftControl) && Input.WasKeyPressed(Keys.D2))
+            {
+                LevelEditor.selectedStructure = StructureType.Spikes;
+            }
+            else if (Input.KeyboardState.IsKeyDown(Keys.LeftControl) && Input.WasKeyPressed(Keys.D3))
+            {
+                //LevelEditor.selectedStructure = StructureType.None;
+            }
+            #endregion
+
+            #region Color Selection
+            // Alt + 1 = White; Alt + 2 = Red; Alt + 3 = Green etc
+            if (Input.KeyboardState.IsKeyDown(Keys.LeftAlt) && Input.WasKeyPressed(Keys.D1))
+            {
+                LevelEditor.selectedColor = Color.White;
+            }
+            else if (Input.KeyboardState.IsKeyDown(Keys.LeftAlt) && Input.WasKeyPressed(Keys.D2))
+            {
+                LevelEditor.selectedColor = Color.Red;
+            }
+            else if (Input.KeyboardState.IsKeyDown(Keys.LeftAlt) && Input.WasKeyPressed(Keys.D3))
+            {
+                LevelEditor.selectedColor = Color.Green;
+            }
+            else if (Input.KeyboardState.IsKeyDown(Keys.LeftAlt) && Input.WasKeyPressed(Keys.D4))
+            {
+                LevelEditor.selectedColor = Color.Blue;
+            }
+            else if (Input.KeyboardState.IsKeyDown(Keys.LeftAlt) && Input.WasKeyPressed(Keys.D5))
+            {
+                LevelEditor.selectedColor = Color.Yellow;
+            }
+            else if (Input.KeyboardState.IsKeyDown(Keys.LeftAlt) && Input.WasKeyPressed(Keys.D6))
+            {
+                LevelEditor.selectedColor = Color.Purple;
+            }
+            else if (Input.KeyboardState.IsKeyDown(Keys.LeftAlt) && Input.WasKeyPressed(Keys.D7))
+            {
+                LevelEditor.selectedColor = Color.Orange;
+            }
+            else if (Input.KeyboardState.IsKeyDown(Keys.LeftAlt) && Input.WasKeyPressed(Keys.D8))
+            {
+                LevelEditor.selectedColor = Color.Pink;
+            }
+            else if (Input.KeyboardState.IsKeyDown(Keys.LeftAlt) && Input.WasKeyPressed(Keys.D9))
+            {
+                LevelEditor.selectedColor = Color.Brown;
+            }
+            else if (Input.KeyboardState.IsKeyDown(Keys.LeftAlt) && Input.WasKeyPressed(Keys.D0))
+            {
+                LevelEditor.selectedColor = Color.Black;
+            }
+            #endregion
+
             LevelEditor.DrawGridLines(Color.Black);
         }
```

Comment:
Nice features. This might be one of the few places where minimal commenting is ok because it's pretty readable.